avocado/utils/process.py

```diff
@@ -1312,12 +1331,12 @@ def system_output(cmd, timeout=None, verbose=True, ignore_status=False,
     :type strip_trail_nl: bool
 
     :return: Command output.
-    :rtype: str
+    :rtype: bytes
     :raise: :class:`CmdError`, if ``ignore_status=False``.
     """
     cmd_result = run(cmd=cmd, timeout=timeout, verbose=verbose, ignore_status=ignore_status,
                      allow_output_check=allow_output_check, shell=shell, env=env,
                      sudo=sudo, ignore_bg_processes=ignore_bg_processes)
     if strip_trail_nl:
-        return cmd_result.stdout.rstrip('\n\r')
+        return cmd_result.stdout.rstrip(b'\n\r')
```

Comment:
Codewise, this is the best implementation. But I'm afraid we are breaking way too many tests out there. Have you considered to make `CmdResult.stdout` to return `stdout.decode(self.encoding)` and create new properties for the raw content, like `CmdResult.stdout_raw`? What would be the downside of that alternative?

Reply:
The downside is that you have perform one extra operation in the vast majority of times.

Also, I thought about the API, and I think that using bytes it's a more honest and explicit one.  I mean, when you an application, I can generate *anything* on STDOUT.  That's the case for instance with the `iso9660` changes in the same patch.  Unless the users *tells* that this is text, and of a given encoding, we shouldn't be in the business of guessing.

And yes, the original proposal on PR #2151 where along the lines you mentioned, but I tried both, and for the reasons mentioned here, I decided to go with a different approach.

Reply:
Oh, I forgot to say: we're not breaking tests because people are using this on Python 2.  And, on Python 2, mixing strings and bytes is OK.  When people start running tests on Python 3, then they need the honest check on how they're handling external (as in coming from processes) data.

Reply:
Fair enough. Let's heavily spread the word about this change.